src/test/java/by/a1qa/testreport/ScreenshotListener.java

```diff
@@ -0,0 +1,35 @@
+package by.a1qa.testreport;
+
+import aquality.appium.mobile.application.AqualityServices;
+import org.apache.commons.io.FileUtils;
+import org.openqa.selenium.OutputType;
+import org.openqa.selenium.TakesScreenshot;
+import org.testng.ITestResult;
+import org.testng.Reporter;
+import org.testng.TestListenerAdapter;
+
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+
+public class ScreenshotListener extends TestListenerAdapter {
+    @Override
+    public void onTestFailure(ITestResult result) {
+        Calendar calendar = Calendar.getInstance();
+        SimpleDateFormat formater = new SimpleDateFormat("dd_MM_yyyy_hh_mm_ss");
+        String dateString = formater.format(calendar.getTime());
+        String methodName = result.getName();
+        if(!result.isSuccess() && AqualityServices.isApplicationStarted()){
+            File scrFile = ((TakesScreenshot)AqualityServices.getApplication().getDriver()).getScreenshotAs(OutputType.FILE);
```

Comment:
this way you will not get screenshot of the full page. Try implement using a library, like `Ashot`, or `Shutterbug`